examples/advanced_books/lib/main.dart

```diff
@@ -1,35 +1,25 @@
 import 'package:flutter/material.dart';
 import 'package:beamer/beamer.dart';
 
-import 'package:advanced_books/locations.dart';
+import 'locations.dart';
 
 void main() => runApp(MyApp());
 
 class MyApp extends StatelessWidget {
-  final notFoundPage = BeamPage(
-    child: Scaffold(
-      body: Center(
-        child: Text('Not found'),
-      ),
+  final routerDelegate = BeamerRouterDelegate(
+    locationBuilder: BeamerLocationBuilder(
```

Comment:
> I just tried out how it would look like using SimpleLocationBuilder instead of BeamerLocationBuilder on another branch.
In my opinion, it looks cleaner and therefore is easier to understand this way because you can directly see each route.
There is still a lot duplicated code (maybe you have an idea how to avoid that), but I like it way more this way.
Let me know what you think :)

It looks cool, but I also very much like that you're using `beamPages.add(...)` in `pagesBuilder` (currently in the PR). It's probably the first example to show this so I would like to keep it.
But... I also agree that routes look good (I don't think we can do anything about duplication as they are so separated).

Therefore I propose, and now you'll finally see the reason why I'm commenting on this line:

Add `final simpleLocationBuilder = ....` that will be configured with `routes` and extract the current `locationBuilder` into `final beamerLocationBuilder = ...`. Then use whichever you prefer and leave a clear comment that these are interchangeable, depending on personal taste (in this case).

Reply:
Sounds good, I will do that and update the PR later or tomorrow. 😊

Reply:
Ok. I'll be pretty busy tomorrow all day so I might merge it on Saturday.
Then this weekend will also be a gread time to publish `v0.12.5`. A lot of cool things have been implemented.

Reply:
Okay, no problem.

I look foward to v0.12.5 😀